### PR TITLE
Fix strong mode issues

### PR DESCRIPTION
**Pull request by dramos07**

### CODE REVIEW

**Comment by dramos07** [comment]
I signed it!

File: lib/src/json_document_transformer.dart [hunk 1]
```diff
@@ -22,7 +22,7 @@ final jsonDocument = new JsonDocumentTransformer();
 /// [FormatException]. If an unencodable object is added to the sink, it
 /// synchronously throws a [JsonUnsupportedObjectError].
 class JsonDocumentTransformer
-    implements StreamChannelTransformer<Object, String> {
+    implements StreamChannelTransformer<dynamic, String> {
```
Comment (nex3) [inline lib/src/json_document_transformer.dart]: Same question as in the other PR: what's the difference between `Object` and `dynamic` here?

Comment (dramos07) [inline lib/src/json_document_transformer.dart]: Without it I get a strong mode errors like the following in 3 different places

[error] The argument type 'JsonDocumentTransformer' cannot be assigned to the parameter type 'StreamChannelTransformer<String, dynamic>'. (package:json_rpc_2/src/client.dart, line 53, col 24)
[error] The argument type 'JsonDocumentTransformer' cannot be assigned to the parameter type 'StreamChannelTransformer<String, dynamic>'. (package:json_rpc_2/src/peer.dart, line 48, col 24)
[error] The argument type 'JsonDocumentTransformer' cannot be assigned to the parameter type 'StreamChannelTransformer<String, dynamic>'. (package:json_rpc_2/src/server.dart, line 60, col 24)

Comment (nex3) [inline lib/src/json_document_transformer.dart]: That looks like an issue that's been fixed in other packages. Does it still reproduce after you run `pub upgrade`?

Comment (dramos07) [inline lib/src/json_document_transformer.dart]: Indeed the latest changes have fixed this. Thanks!

**Comment by dramos07** [comment]
The fix is no longer needed.

**Pull request closed without merging**